<?xml version="1.0"?>
<!--
    Zed based odometry launch file
-->

<launch>
  <arg name="xavier" default="false"/>

  <!-- IMU -->
  <arg name="i2c_bus" value="/dev/i2c-8" if="$(arg xavier)"/>
  <arg name="i2c_bus" value="/dev/i2c-1" unless="$(arg xavier)"/>
  <node ns="imu" name="imu_node" pkg="svea_sensors" type="bno055_i2c_node" respawn="true" respawn_delay="2">
    <param name="device" type="string" value="$(arg i2c_bus)"/>
    <param name="address" type="int" value="40"/> <!-- 0x28 == 40 is the default for BNO055 -->
    <param name="frame_id" type="string" value="imu"/>
  </node>
  <!-- Static transforms for sensors -->
  <include file="$(find svea_sensors)/launch/transforms.launch">
    <arg name="xavier" value="$(arg xavier)"/>
  </include>
  
  <!-- Zed cam -->
  <arg name="node_name"            default="zed_node" />
  <arg name="publish_urdf"         default="true" />
  <group ns="zed">
    <include file="$(find zed_wrapper)/launch/zed_no_tf.launch">
        <arg name="node_name"           value="$(arg node_name)" />
        <arg name="camera_model"        value="zedm" />
        <arg name="publish_urdf"        value="$(arg publish_urdf)" />
        <arg name="cam_pos_x"           value="0.08" />
        <arg name="cam_pos_y"           value="0.015" />
        <arg name="cam_pos_z"           value="0.21" />
        <arg name="cam_roll"            value="0.0" />
        <arg name="cam_pitch"           value="1.507" />
        <arg name="cam_yaw"             value="3.14" />
    </include>
  </group>
  
  <!-- Robot localization-->
  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
    <rosparam command="load" file="$(find svea_sensors)/params/robot_localization/zed_ekf.yaml" />
    <param name="imu0" value="/imu/data"/>
    <param name="odom0" value="/zed/zed_node/odom"/>
    <!--  Placeholder for output topic remapping
    <remap from="odometry/filtered" to=""/>
    <remap from="accel/filtered" to=""/>
    -->
  </node>
  
  <!-- Hokuyo LIDAR -->
  <arg name="ip_address" default="192.168.3.11" /> <!-- <arg name="svo_file" default="path/to/svo/file.svo"> -->
  <node pkg="urg_node" type="urg_node" name="Hokyoulidar">
    <param name="ip_address"    value="$(arg ip_address)" />
    <param name="frame_id" value="laser"/>
    <param name="calibrate_time" value="true"/>
    <param name="publish_intensity" value="true"/>
    <param name="publish_multiecho" value="false"/>
    <param name="angle_min" value="-2.355"/>
    <param name="angle_max" value="2.355"/>
  </node>
</launch>
